docs/DeveloperGuide.adoc

```diff
@@ -364,6 +387,144 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+* Sorting
+
+==== Favorite class
+A favorite class is implemented and used by Person class. A person object does not construct with a favorite object but it is
+able to setFavorite a new favorite object.
+A favorite object is different from other attribute object is that it just contains a simple boolean property.
+
+To favorite a person in the list, a favorite command is implemented. The favorite command follows the same parser logic
+as `delete` command, which is to parse the index of the person to favorite. Hence, the favoriteCommand also extends
+UndoableCommand and overrides the methods in UndoableCommand.
+
+Model and UI components also have to be updated. For model, a method toggle favorite is implemented to favorite or "defavorite"
+a person at specified index:
+
+[source,java]
+----
+public void favorite(ReadOnlyPerson toFavorite)
+    throws PersonNotFoundException {
+        requireNonNull(toFavorite);
+        int index = internalList.indexOf(toFavorite);
+
+        if (index == -1) {
+            throw new PersonNotFoundException();
+        }
+
+        Person newPerson = new Person(toFavorite);
+        newPerson.setFavorite(new Favorite(!toFavorite.getFavorite().favorite));
+
+        internalList.set(index, newPerson);
+        sort();
+}
+----
+
+And for UI, a simple label is created and `setVisible` to `true` for any index that is
+favorited. The labels are `setGraphic` to followed image:
+
+image::heart.png[width="40"]
+
+==== Sorting
+
+A sorting algorithm needs to be implemented for the `UniquePersonList` of the `Model` Component.
+Fortunately, it can be easily done with `List.sort()` and  `Java predicate`. The priority of sorting is such that
+the persons favorited always appear on the top of the list, then all persons w/out favorited are
+sorted by alphabetical order:
+
+[source,java]
+----
+private void sort() {
+        internalList.sort((ReadOnlyPerson p1, ReadOnlyPerson p2) -> {
+            if (!p1.getFavorite().equals(p2.getFavorite())) {
+                return p2.getFavorite().getValue() - p1.getFavorite().getValue();
+            } else {
+                return p1.getName().fullName.compareTo(p2.getName().fullName);
+            }
+        });
+}
+----
+
+Finally, this method needs to be called whenever an event that causes changes to the model is called, eg
+Edit a person, Add a person or Favorite a person.
+
+
+==== Design Considerations
+
+**Aspect:** Where sorting should occurs +
+**Alternative 1 (current choice):** On the lowest layer of `Model`, which is `UniquePersonList` +
+**Pros:** No need to have sorting in upper layer. +
+**Cons:** Permanently mutate the data list. +
+**Alternative 2:** Only sorting on the surface of `UI` component +
+**Pros:** No mutation of data list. +
+**Cons:** Difficult to implement system tests. +
+// end::favorite[]
+
+
+== Birthday Detail
+
+A birthday detail is implemented under the Person class as Birthday. Users will add the birthday in along with the add command.
```

Comment:
This is hardly an implementation description! It's merely a description of your functionality. You should be describing how your functionality is implemented, and how it works when called. See the undo/redo implementation details for an example.